src/backend/distributed/utils/node_metadata.c

```diff
@@ -345,6 +346,7 @@ AddNodeMetadata(char *nodeName, int32 nodePort, int32 groupId, char *nodeRack,
 	char *nodeInsertCommand = NULL;
 	List *workerNodeList = NIL;
 
+	EnsureSchemaNode();
```

Comment:
For `master_add_node`,  `master_initialize_node_metadata` and `create_reference_table`, the UDF definition itself is just a thin wrapper around the actual logic and other checks are done in the actual function itself. I tried to put the schema node check into where all other checks (like superuser check) are done.